test/language/expressions/async-generators/throw-suspendedYield.js

```diff
@@ -0,0 +1,39 @@
+// Copyright 2017 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+author: Caitlin Potter <caitp@igalia.com>
+esid: pending
+description: >
+  AsyncGeneratorResumeNext:
+  If completion.[[Type]] is throw, and generator.[[AsyncGeneratorState]] is
+  "suspendedYield", generator is resumed and immediately and
+  closes the generator and returns completion.
+flags: [async]
+---*/
+
+class Err extends Error {}
+var g = async function*() {
+  yield 1;
+  $ERROR('Generator must not be resumed.');
+};
+
+var it = g();
+it.next().then(function(ret) {
+  // Initial yield
+  assert.sameValue(ret.value, 1);
+  assert.sameValue(ret.done, false);
```

Comment:
the comments are great to track the steps. If you don't mind, can you move them to become assertion messages?

This way we can have a powerful track of each assertion and, better, tells the implementors where each test is failing at.

Example for the case above:

```js
assert.sameValue(ret.value, 1, 'Initial yield');
assert.sameValue(ret.done, false, 'Initial yield');
```

Even if they are the same message, I can already identify each one due to the respective expected values.